test/e2e/velero_utils.go

```diff
@@ -5,14 +5,112 @@ import (
 	"encoding/json"
 	"fmt"
 	"io"
+	"os"
 	"os/exec"
+	"path/filepath"
 
 	"github.com/pkg/errors"
 
-	v1 "github.com/vmware-tanzu/velero/pkg/apis/velero/v1"
+	velerov1api "github.com/vmware-tanzu/velero/pkg/apis/velero/v1"
+	"github.com/vmware-tanzu/velero/pkg/client"
+	cliinstall "github.com/vmware-tanzu/velero/pkg/cmd/cli/install"
+	"github.com/vmware-tanzu/velero/pkg/cmd/util/flag"
+	"github.com/vmware-tanzu/velero/pkg/install"
 )
 
-func CheckBackupPhase(ctx context.Context, veleroCLI string, backupName string, expectedPhase v1.BackupPhase) error {
+func getProviderPlugins(providerName string) []string {
+	// TODO: make plugin images configurable
+	switch providerName {
+	case "aws":
+		return []string{"velero/velero-plugin-for-aws:v1.1.0"}
+	case "azure":
+		return []string{"velero/velero-plugin-for-microsoft-azure:v1.1.1"}
+	case "vsphere":
+		// TODO: find correct image tag
+		return []string{"velero/velero-plugin-for-vsphere:v1.1.1"}
+	default:
+		return []string{""}
+	}
+}
+
+// GetProviderVeleroInstallOptions returns Velero InstallOptions for the provider.
+func GetProviderVeleroInstallOptions(providerName, credentialsFile, objectStoreBucket, objectStorePrefix string,
+	bslConfig, vslConfig string,
+	plugins []string,
+) (*cliinstall.InstallOptions, error) {
+
+	if credentialsFile == "" {
+		return nil, errors.Errorf("No credentials were supplied to use for E2E tests")
+	}
+
+	realPath, err := filepath.Abs(credentialsFile)
+	if err != nil {
+		return nil, err
+	}
+
+	io := cliinstall.NewInstallOptions()
+	// always wait for velero and restic pods to be running.
+	io.Wait = true
+	io.ProviderName = providerName
+	io.SecretFile = credentialsFile
+
+	io.BucketName = objectStoreBucket
+	io.Prefix = objectStorePrefix
+	io.BackupStorageConfig = flag.NewMap()
+	io.BackupStorageConfig.Set(bslConfig)
+
+	io.VolumeSnapshotConfig = flag.NewMap()
+	io.VolumeSnapshotConfig.Set(vslConfig)
+
+	io.SecretFile = realPath
+	io.Plugins = flag.NewStringArray(plugins...)
+	return io, nil
+}
+
+// InstallVeleroServer installs velero in the cluster using `velero install command`
+func InstallVeleroServer(ctx context.Context, io *cliinstall.InstallOptions) error {
+	config, err := client.LoadConfig()
+	if err != nil {
+		return err
+	}
+
+	vo, err := io.AsVeleroOptions()
+	if err != nil {
+		return errors.Wrap(err, "Failed to translate InstallOptions to VeleroOptions for Velero")
+	}
+
+	f := client.NewFactory("e2e", config)
+	resources, err := install.AllResources(vo)
```

Comment:
I've had a test fail here because the `velero` namespace was in `Terminating` state from a previous run. Perhaps the `AfterEach` hooks where the namespace is deleted could be made to wait.

Reply:
Yes, this is listed as a limitation in the docs. Going to fix that in a follow-up PR.

Reply:
Ah, I missed that. Sorry!